nifi-nar-bundles/nifi-kudu-bundle/nifi-kudu-processors/src/main/java/org/apache/nifi/processors/kudu/AbstractKudu.java

```diff
@@ -0,0 +1,232 @@
+            .required(true)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .build();
+
+    protected static final PropertyDescriptor TABLE_NAME = new PropertyDescriptor.Builder()
+            .name("Table Name")
+            .description("The name of the Kudu Table to put data into")
+            .required(true)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .build();
+
+    public static final PropertyDescriptor RECORD_READER = new PropertyDescriptor.Builder()
+            .name("record-reader")
+            .displayName("Record Reader")
+            .description("The service for reading records from incoming flow files.")
+            .identifiesControllerService(RecordReaderFactory.class)
+            .required(true)
+            .build();
+
+    protected static final PropertyDescriptor SKIP_HEAD_LINE = new PropertyDescriptor.Builder()
+            .name("Skip head line")
+            .description("Set it to true if your first line is the header line e.g. column names")
+            .allowableValues("true", "false")
+            .defaultValue("true")
+            .required(true)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .build();
+
+    protected static final PropertyDescriptor INSERT_OPERATION = new PropertyDescriptor.Builder()
+            .name("INSERT OPERATION")
+            .description("Specify operation for this processor. Insert-Ignore will ignore duplicated rows")
+            .allowableValues("Insert", "Insert-Ignore", "Upsert")
+            .defaultValue("Insert")
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .build();
+
+    protected static final Relationship REL_SUCCESS = new Relationship.Builder()
+            .name("success")
+            .description("A FlowFile is routed to this relationship after it has been successfully stored in Kudu")
+            .build();
+    protected static final Relationship REL_FAILURE = new Relationship.Builder()
+            .name("failure")
+            .description("A FlowFile is routed to this relationship if it cannot be sent to Kudu")
+            .build();
+
+    public static final String RECORD_COUNT_ATTR = "record.count";
+
+    protected String kuduMasters;
+    protected String tableName;
+    protected boolean skipHeadLine;
+    protected String operation;
+
+    protected KuduClient kuduClient;
+    protected KuduTable kuduTable;
+
+    @OnScheduled
+    public void onScheduled(final ProcessContext context) {
+        try {
+            tableName = context.getProperty(TABLE_NAME).getValue();
+            kuduMasters = context.getProperty(KUDU_MASTERS).getValue();
+            skipHeadLine = context.getProperty(SKIP_HEAD_LINE).asBoolean();
+            operation = context.getProperty(INSERT_OPERATION).getValue();
+
+            kuduClient = getKuduConnection(kuduMasters);
```

Comment:
Since @toddlipcon confirmed that Kudu uses lazy connections, we should instead revert back to using `OnEnabled` so that we only create the connection once when the process is enabled.